lib/Sema/CSRanking.cpp

```diff
@@ -1334,7 +1334,7 @@ SolutionDiff::SolutionDiff(ArrayRef<Solution> solutions) {
             overloadChoice.first,
             overloadChoice.second
           });
-        
+        break;
```

Comment:
I wonder if we should be actually breaking here and if the case for `typeBindings` is actually correct because it seems to me when evaluating solutions we are supposed to be considering all of the pairs with differences (cartesian product excluding equal pairs) and not just first encountered? E.g. If we have solutions:  1 (with A), 2 (with B) and 3 (with C) for the same locator, `compareSolutions` should be checking all of the different pairs and not just `(A, B)`, right?

Reply:
Ok, I've looked a bit closer and the break here is valid because SolutionDiff is going to include the whole set of overloads.